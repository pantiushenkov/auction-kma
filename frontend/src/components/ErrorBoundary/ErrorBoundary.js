import React, { Component } from 'react'

export default class ErrorBoundary extends Component {
  constructor(props) {
    super(props)
    this.state = {
      hasError: false,
      errorMessage: '',
    }
  }

  componentDidCatch(error) {
    this.setState({
      hasError: true,
      errorMessage: error.message,
    })
    // You can also log the error to an error reporting service
    console.log('CAUGHT ERROR:', error, this.state)
  }

  render() {
    const { hasError, errorMessage } = this.state
    if (hasError && errorMessage) {
      return <div>Sorry, error occured: {errorMessage}</div>
    }
    return this.props.children
  }
}
