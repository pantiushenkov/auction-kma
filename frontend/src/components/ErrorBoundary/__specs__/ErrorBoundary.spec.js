import React from 'react'
import { shallow } from 'enzyme'
import ErrorBoundary from '../ErrorBoundary'

jest.mock('../../Stackdriver/errorHandler', () => {
  return {
    report: jest.fn()
  }
})

describe('<ErrorBoundary />', () => {
  let wrapper,
    children
  beforeEach(() => {
    children = <div>Test</div>
    wrapper = shallow(
      <ErrorBoundary
        children={children}
      />
    )
  })
  it('should render this.props.children', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call setState in componentDidCatch', () => {
    wrapper.instance().setState = jest.fn()
    wrapper.instance().componentDidCatch({
      message: 'error'
    })
    expect(wrapper.instance().setState).toBeCalledWith({'errorMessage': 'error', 'hasError': true})
  })

  it('should render Sorry, error occured', () => {
    wrapper.instance().setState({
      hasError: true,
      errorMessage: true
    })
    expect(wrapper.update()).toMatchSnapshot()
  })
})
