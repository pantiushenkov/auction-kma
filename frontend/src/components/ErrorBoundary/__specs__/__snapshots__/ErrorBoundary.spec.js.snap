// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary /> should render Sorry, error occured 1`] = `
<div>
  Sorry, error occured: 
</div>
`;

exports[`<ErrorBoundary /> should render this.props.children 1`] = `
<div>
  Test
</div>
`;
